src/sensor_state_data/data.py

```diff
@@ -279,3 +285,27 @@ def update_sensor(
             native_unit_of_measurement=native_unit_of_measurement,
             device_class=device_class,
         )
+
+    def update_event(
```

Comment:
Should the name be `fire_event`, like the doc string says it does?

Reply:
yes, sound good to me. Changed it.